src/SAD/Import/Reader.hs

```diff
@@ -46,62 +51,84 @@ instructionFile = after (optLL1 [] $ chainLL1 instr) eof
 
 -- Reader loop
 
-readProofText :: Text -> [ProofText] -> IO [ProofText]
-readProofText pathToLibrary text0 = do
+-- | @readProofText startWithTex pathToLibrary text0@ takes:
+-- @startWithTex@, a boolean indicating whether to execute the next file instruction using the tex parser,
+-- @pathToLibrary@, a path to where the read instruction should look for files and
+-- @text0@, containing some configuration.
+readProofText :: Bool -> Text -> [ProofText] -> IO [ProofText]
+readProofText startWithTex pathToLibrary text0 = do
   pide <- Message.pideContext
-  (text, reports) <- reader pathToLibrary [] [State (initFS pide) noTokens noSourcePos] text0
+  let initialParserKind = if startWithTex then Tex else NonTex
+  (text, reports) <- reader pathToLibrary [] [State (initFS pide) noTokens NonTex noSourcePos] text0 (Just initialParserKind)
   when (isJust pide) $ Message.reports reports
   return text
 
-reader :: Text -> [Text] -> [State FState] -> [ProofText] -> IO ([ProofText], [Message.Report])
+reader :: Text -> [Text] -> [State FState] -> [ProofText] -> Maybe ParserKind -> IO ([ProofText], [Message.Report])
 reader pathToLibrary doneFiles = go
   where
-    go stateList [ProofTextInstr pos (GetArgument Read file)] = if ".." `Text.isInfixOf` file
+    go stateList [ProofTextInstr pos (GetArgument Read file)] _ = if ".." `Text.isInfixOf` file
+      then Message.errorParser (Instr.position pos) ("Illegal \"..\" in file name: " ++ show file)
+      else go stateList [ProofTextInstr pos $ GetArgument File $ pathToLibrary <> "/" <> file] (Just NonTex)
+    
+    go stateList [ProofTextInstr pos (GetArgument ReadTex file)] _ = if ".." `Text.isInfixOf` file
       then Message.errorParser (Instr.position pos) ("Illegal \"..\" in file name: " ++ show file)
-      else go stateList [ProofTextInstr pos $ GetArgument File $ pathToLibrary <> "/" <> file]
+      else go stateList [ProofTextInstr pos $ GetArgument File $ pathToLibrary <> "/" <> file] (Just Tex)
 
-    go (pState:states) [ProofTextInstr pos (GetArgument File file)]
+    go (pState:states) [ProofTextInstr pos (GetArgument File file)] parserKind'
       | file `elem` doneFiles = do
+          when (Just (parserKind pState) /= parserKind')
```

Comment:
I am slightly confused by this, could you add a comment why this condition corresponds to the error message below?

Reply:
I added a comment explaining how choosing the parserKind works and why this error message is justified. Also improved the error message in order to be more clear about what is going on.